Uptime.CredentialManager.Web/Controllers/CredentialsController.cs

```diff
@@ -46,9 +47,35 @@ public IEnumerable<SelectListItem> GetUsers(Expression<Func<User, bool>> predica
         // GET: Credentials
         public async Task<IActionResult> Index()
         {
-            return View(await _context.Credential.ToListAsync());
+            if (User.Identity.IsAuthenticated)
```

Comment:
Hmmm, usually you would use the general [Authorize] attribute on top of action or even controller and this makes sure you have the User object available that you can get username from. No need to check the IsAuthenticated in the action body.